std/c/dragonfly.zig

```diff
@@ -0,0 +1,1788 @@
+extern "c" threadlocal var errno: c_int;
+pub fn _errno() *c_int {
+    return &errno;
+}
+pub const __sighandler_t = extern fn(c_int) void;
+// auto import
```

Comment:
The auto import brought in some undesirable stuff, such as `NULL` and probably more that isn't DragonFly OS specific. Due to zig's lazy analysis however this is mostly fine, as unused things are generally harmless.

Types and constants should be moved to std/os/bits/dragonfly.zig. The extern "c" functions should stay.